cluster-autoscaler/simulator/clustersnapshot/delta.go

```diff
@@ -402,6 +403,31 @@ func NewDeltaClusterSnapshot() *DeltaClusterSnapshot {
 	return snapshot
 }
 
+func (snapshot *DeltaClusterSnapshot) GetNodeInfo(nodeName string) (*framework.NodeInfo, error) {
+	schedNodeInfo, err := snapshot.getNodeInfo(nodeName)
+	if err != nil {
+		return nil, err
+	}
+	return framework.WrapSchedulerNodeInfo(schedNodeInfo), nil
+}
+
+func (snapshot *DeltaClusterSnapshot) ListNodeInfos() ([]*framework.NodeInfo, error) {
+	schedNodeInfos := snapshot.data.getNodeInfoList()
+	return framework.WrapSchedulerNodeInfos(schedNodeInfos), nil
+}
+
+func (snapshot *DeltaClusterSnapshot) AddNodeInfo(nodeInfo *framework.NodeInfo) error {
```

Comment:
Why have this method at all (here and elsewhere) if you're discarding all the extra data in framework.NodeInfo and just storing the scheduler representation? I'd expect the result of:
```
snapshot.AddNodeInfo(x)
y = snapshot.GetNodeInfo(x.Node.Name)
```
To be that x and y are the same. I think anyone would instinctively assume the same, but that's not the case here and that strikes me as confusing and ultimately error prone when someone makes a wrong assumption. If you only want to store scheduler stuff, I'd rather this method didn't exist at all and everyone would just keep using AddNodeWithPods. 
Even if that requires a tiny bit of boilerplate to unpack NodeInfo it is much more clear what's happening here. I know ResourceClaims/Slices will not be stored if I can't even pass them. If I do pass them to snapshot, I expect it will store them as well.

Reply:
The extra data will be stored and your example will hold, this just comes in later commits (https://github.com/kubernetes/autoscaler/pull/7350/commits/ef9d4209186909a1a28ef60b7cc9d107d6631c07#diff-29296088fba933fa837b2efdce93b2423b19247d99f59f1d8ef1bc8d5e8c6915). 

I can remove the DRA fields from the new types until then, but then we have wrapper types for seemingly no reason - either way isn't ideal in the intermediate, but the end state should be clear IMO.